lib/hammer_cli/defaults.rb

```diff
@@ -0,0 +1,207 @@
+require 'hammer_cli'
+require 'yaml'
+module HammerCLI
+
+  class BaseDefaultsProvider
+    def self.plugin_name
+      self.name.split('::').first.gsub(/^HammerCLI/, '').underscore
+    end
+
+    def self.register_provider
+      HammerCLI.defaults.register_provider(self)
+    end
+
+    def self.support?
+      raise NotImplementedError
+    end
+
+    def self.get_defaults
+      raise NotImplementedError
+    end
+  end
+
+  class DefaultsCommand < HammerCLI::AbstractCommand
+
+    class ListDefaultsCommand < HammerCLI::DefaultsCommand
+      command_name 'list'
+
+      desc _('List all the default parameters')
+      def execute
+        list_all_defaults_message
+        HammerCLI::EX_OK
+      end
+    end
+
+    class DeleteDefaultsCommand < HammerCLI::DefaultsCommand
+      command_name 'delete'
+
+      desc _('Delete a default param')
+      option "--param-name", "OPTION_NAME", _("The name of the default option"), :required => true
+
+      def execute
+        if context[:defaults].defaults_settings && context[:defaults].defaults_settings[option_param_name.to_sym]
+          context[:defaults].delete_default_from_conf(option_param_name.to_sym)
+          param_deleted(option_param_name)
+        else
+          variable_not_found
+        end
+        HammerCLI::EX_OK
+      end
+    end
+
+    class AddDefaultsCommand < HammerCLI::DefaultsCommand
+      command_name 'add'
+
+      desc _('Add a default parameter to config')
+      option "--param-name", "OPTION_NAME", _("The name of the default option"), :required => true
+      option "--param-val", "OPTION_VALUE", _("The value for the default option")
+      option "--plugin-name", "OPTION_PLUGIN_NAME", _("The plugin name defaults will be generated for")
```

Comment:
--provider-name,  _("The name of the provider providing the value. For list available providers see hammer defaults providers.")